client.go

```diff
@@ -510,6 +519,23 @@ func Connect(ctx context.Context, conn net.Conn, opt Options) (*Client, error) {
 		compression = proto.CompressionDisabled
 	}
 
+	// Handle SSH authentication
+	var sshKey *ssh.SSHKey
+	var useSSH bool
+	if opt.SSHKeyFile != "" {
+		var err error
+		sshKey, err = ssh.LoadPrivateKeyFromFile(opt.SSHKeyFile, opt.SSHKeyPassphrase)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to load SSH key")
+		}
+		useSSH = true
+	}
+
+	user := opt.User
+	if useSSH {
```

Comment:
This variable is excessive, we can use `if opt.SSHSigner != nil` here.

Reply:
done